components/camera/camera.go

```diff
@@ -171,6 +161,100 @@ func DecodeImageFromCamera(ctx context.Context, mimeType string, extra map[strin
 	return img, nil
 }
 
+// GetImageFromGetImages is a utility function to quickly implement GetImage from an already-implemented GetImages method.
+// It returns a byte slice and ImageMetadata, which is the same response signature as the Image method.
+//
+// If sourceName is nil, it returns the first image in the response slice.
+// If sourceName is not nil, it returns the image with the matching source name.
+// If no image is found with the matching source name, it returns an error.
+//
+// It uses the mimeType arg to specify how to encode the bytes returned from GetImages.
+func GetImageFromGetImages(ctx context.Context, sourceName *string, mimeType string, cam Camera) ([]byte, ImageMetadata, error) {
+	// TODO(RSDK-10991): pass through extra field when implemented
+	images, _, err := cam.Images(ctx)
+	if err != nil {
+		return nil, ImageMetadata{}, fmt.Errorf("could not get images from camera: %w", err)
+	}
+	if len(images) == 0 {
+		return nil, ImageMetadata{}, errors.New("no images returned from camera")
+	}
+
+	// if mimeType is empty, use JPEG as default
+	if mimeType == "" {
+		mimeType = utils.MimeTypeJPEG
+	}
+
+	var img image.Image
+	if sourceName == nil {
+		img = images[0].Image
+	} else {
+		for _, i := range images {
+			if i.SourceName == *sourceName {
+				img = i.Image
+				break
+			}
+		}
+		if img == nil {
+			return nil, ImageMetadata{}, errors.New("no image found with source name: " + *sourceName)
+		}
+	}
+
+	if img == nil {
+		return nil, ImageMetadata{}, errors.New("image is nil")
+	}
+
+	imgBytes, err := rimage.EncodeImage(ctx, img, mimeType)
+	if err != nil {
+		return nil, ImageMetadata{}, fmt.Errorf("could not encode image: %w", err)
+	}
+	return imgBytes, ImageMetadata{MimeType: mimeType}, nil
+}
+
+// GetImagesFromGetImage is a utility function to quickly implement GetImages from an already-implemented GetImage method.
+// It takes a mimeType and a camera as args, and returns a slice of NamedImage and ResponseMetadata,
+// which is the same response signature as the Images method. We use the mimeType arg to specify
+// how to decode the image bytes returned from GetImage. Source name is empty string always.
+// It returns a slice of NamedImage of length 1 and ResponseMetadata, using the camera's name as the source name.
+func GetImagesFromGetImage(
+	ctx context.Context,
+	mimeType string,
+	cam Camera,
+	logger logging.Logger,
+) ([]NamedImage, resource.ResponseMetadata, error) {
+	// TODO(RSDK-10991): pass through extra field when implemented
+	resBytes, resMetadata, err := cam.Image(ctx, mimeType, nil)
+	if err != nil {
+		return nil, resource.ResponseMetadata{}, fmt.Errorf("could not get image bytes from camera: %w", err)
+	}
+	if len(resBytes) == 0 {
+		return nil, resource.ResponseMetadata{}, errors.New("received empty bytes from camera")
+	}
+
+	resMimetype, _ := utils.CheckLazyMIMEType(resMetadata.MimeType)
+	if resMimetype != mimeType {
```

Comment:
should we also strip lazy from `mimeType` so that this check doesn't trigger if mimeType has the lazy suffix but is otherwise the same as the response?